heron/api/src/scala/com/twitter/heron/streamlet/scala/impl/StreamletImpl.scala

```diff
@@ -141,7 +145,14 @@ class StreamletImpl[R](
     *
     * @param numClones The number of clones to clone
     */
-  override def clone(numClones: Int): Seq[Streamlet[R]] = ???
+  override def clone(numClones: Int): Seq[Streamlet[R]] = {
```

Comment:
can this become an perf issue if the number of clones are high?

Reply:
Currently, there is no limitation about the `number of clones` and `number of partitions`. I think limitation can be useful as long-term by getting team-agreement first and putting Java tier.